main/src/ca/uwaterloo/flix/language/ast/ParsedAst.scala

```diff
@@ -602,6 +602,117 @@ object ParsedAst {
       */
     case class Tuple(sp1: SourcePosition, elms: Seq[ParsedAst.Expression], sp2: SourcePosition) extends ParsedAst.Expression
 
+    /**
+      * ArrayLit Expression.
+      *
+      * @param sp1 the position of the first character in the expression.
+      * @param elms the elements of the array.
+      * @param sp2 the position of the last character in the expression.
+      */
+    case class ArrayLit(sp1: SourcePosition, elms: Seq[ParsedAst.Expression], sp2: SourcePosition) extends ParsedAst.Expression
+
+    /**
+      * ArrayNew Expression
+      *
+      * @param sp1 the position of the first character in the expression.
+      * @param elm the default value of the array elements.
+      * @param len the length of the array.
+      * @param sp2 the position of the last character in the expression.
+      */
+    case class ArrayNew(sp1: SourcePosition, elm: ParsedAst.Expression, len: ParsedAst.Expression, sp2: SourcePosition) extends  ParsedAst.Expression
+
+    /**
+      * ArrayLoad Expression
+      *
+      * @param exp1 the array.
+      * @param exp2 the index of the array.
+      * @param sp2 the position of the last character in the expression.
+      */
+    case class ArrayLoad(exp1: ParsedAst.Expression, exp2: ParsedAst.Expression, sp2: SourcePosition) extends  ParsedAst.Expression
+
+    /**
+      * ArrayStore Expression
+      * @param exp1 the array.
+      * @param exp2 the index of the array.
+      * @param exp3 the element to be inserted at the given index.
+      * @param sp2 the position of the last character in the expression.
+      */
+    case class ArrayStore(exp1: ParsedAst.Expression, exp2: ParsedAst.Expression, exp3: ParsedAst.Expression, sp2: SourcePosition) extends  ParsedAst.Expression
+
+    /**
+      * ArrayLenght Expression
+      *
+      * @param sp1 the position of the first character in the expression.
+      * @param exp the array
+      * @param sp2 the position of the last character in the expression.
+      */
+    case class ArrayLength(sp1: SourcePosition, exp: ParsedAst.Expression, sp2: SourcePosition) extends ParsedAst.Expression
+
+    /**
+      * ArraySlice Expression
+      *
+      * @param exp1 the array
+      * @param exp2 the first index
+      * @param exp3 the second index
+      * @param sp2 the position of the last character in the expression.
+      */
+    case class ArraySlice(exp1: ParsedAst.Expression, exp2: ParsedAst.Expression, exp3: ParsedAst.Expression, sp2: SourcePosition) extends ParsedAst.Expression
+
+    /**
+      * VecLit Expression.
+      *
+      * @param sp1 the position of the first character in the expression.
+      * @param elms the elements of the vector.
+      * @param sp2 the position of the last character in the expression.
+      */
+    case class VecLit(sp1: SourcePosition, elms: Seq[ParsedAst.Expression], sp2: SourcePosition) extends ParsedAst.Expression
```

Comment:
Maybe we should call all these Vector, i.e. VectorLit, VectorNew, etc. since we also write ArrayNew etc. To make it consistent.